pkgs/tools/security/pynitrokey/default.nix

```diff
@@ -1,64 +1,77 @@
 { lib
-, python3Packages
+, buildPythonApplication
 , fetchPypi
-, nrfutil
-, libnitrokey
 , nix-update-script
+, installShellFiles
+, libnitrokey
+, flit-core
+, certifi
+, cffi
+, click
+, cryptography
+, ecdsa
+, frozendict
+, fido2
+, intelhex
+, nkdfu
+, python-dateutil
+, pyusb
+, requests
+, spsdk
+, tqdm
+, tlv8
+, typing-extensions
+, pyserial
+, protobuf
+, click-aliases
+, semver
+, nethsm
+, importlib-metadata
 }:
 
-with python3Packages;
-
 buildPythonApplication rec {
   pname = "pynitrokey";
-  version = "0.4.40";
-  format = "pyproject";
+  version = "0.4.41";
+  pyproject = true;
 
   src = fetchPypi {
     inherit pname version;
-    hash = "sha256-Hu+8UooDzv4GhkWt0sCckQQyHjWn4V/zt2ADlVCoHmk=";
+    hash = "sha256-QszxzZ2I/Bc7stcUh02w6JW/UlF9l77ZyaeWP9TXhPY=";
   };
 
   propagatedBuildInputs = [
     certifi
     cffi
     click
-    click-aliases
     cryptography
     ecdsa
     frozendict
     fido2
     intelhex
     nkdfu
-    nrfutil
     python-dateutil
     pyusb
     requests
-    semver
     spsdk
     tqdm
-    urllib3
     tlv8
     typing-extensions
+    pyserial
+    protobuf
+    click-aliases
+    semver
+    nethsm
     importlib-metadata
   ];
 
   nativeBuildInputs = [
     flit-core
-    pythonRelaxDepsHook
+    installShellFiles
   ];
 
-  # FIXME: does pythonRelaxDepsHook not work for pypaBuildHook + flit-core?
+  # pythonRelaxDepsHook runs in postBuild so cannot be used
   pypaBuildFlags = [ "--skip-dependency-check" ];
 
-  pythonRelaxDeps = [
-    "click"
-    "cryptography"
-    "protobuf"
-    "python-dateutil"
-    "spsdk"
-    "typing_extensions"
-  ];
-
   # libnitrokey is not propagated to users of the pynitrokey Python package.
```

Comment:
```suggestion
  pythonRelaxDeps = [
    "click"
    "cryptography"
    "protobuf"
    "python-dateutil"
    "spsdk"
    "typing_extensions"
  ];
  
  # libnitrokey is not propagated to users of the pynitrokey Python package.
```

Reply:
Oh I see, I also had the same misunderstanding of what `pythonRelaxDepsHook` does. Now I'

Reply:
In general I prefer maintaining an explicit list, but at the moment when upstream still changes frequently, maybe `pythonRelaxDeps = true;` is also acceptable. (you may resolve my other comments, they are fine)